### PR TITLE
Display with object & interactivity

### PR DESCRIPTION
**Pull request by JLBegin**
## Changes
- new method ImagingPath.displayWithObject(diameter) to explicitly set an object to draw, while ImagingPath.display defaults to the FOV view with principal and axial rays. 
- creating graphics from the provided raysList
- changed FOV display: showing P&A rays + grayed out outlined arrows of FOV object and its conjugate planes
- Interactive mode with check-boxes to select which 'group' to show (FOV, object, lamp, etc.)

## Examples
Using demo 5 of the examples:
```
path = ImagingPath()
path.append(Space(d=4))
path.append(Lens(f=4, diameter=0.8, label='Obj'))
path.append(Space(d=4 + 18))
path.append(Lens(f=18, diameter=5.0, label='Tube Lens'))
path.append(Space(d=18))
```
### path.display()
Default to onlyPrincipalAndAxialRays. 
image

### path.display(rays=ObjectRays(diameter=0.8, halfAngle=0.05))
image

### path.displayWithObject(diameter=2)
image

### path.displayWithObject(diameter=0.8, fanAngle=0.05)
image

### path.displayWithObject(diameter=0.8, fanAngle=0.05, removeBlocked=False)
image

## display LampRays
```
path.design(rayColors=['b', 'g'])
path.display(rays=LampRays(diameter=1, NA=0.01, N=50))
```
Currently choosing not to display any images. 

image

### CODE REVIEW

**Comment by JLBegin** [comment]
What it looks like for now in interactive mode: 

No object provided, a default object is still created to fit FOV.
```
path.display(raysList=[LampRays(diameter=1, NA=0.01, N=50))
```
RT_interactive

If object provided:
```
path.display(raysList=[LampRays(diameter=1, NA=0.01, N=50), ObjectRays(1.5, 0.05)], removeBlocked=False)
```
ezgif com-video-to-gif

File: raytracing/figure.py [hunk 1]
```diff
@@ -12,27 +13,43 @@ class Figure:
     """
     def __init__(self, opticalPath):
         self.path = opticalPath
+        self.raysList = []
 
-        self.graphics = []
-        self.lines = []
+        self.graphicGroups = {'FOV': [], 'object': [], 'lamp': [], 'elements': []}
```
Comment (dccote) [inline raytracing/figure.py:18]: FOV should be renamed. We really mean "principal and axial rays from FOV".
Object should be: "Object-Image" I think?

File: raytracing/figure.py [hunk 2]
```diff
@@ -467,12 +545,20 @@ def create(self, comments=None, title=None):
...
         self.draw()
 
         self.axes.callbacks.connect('ylim_changed', self.onZoomCallback)
         plt.connect('resize_event', self.onZoomCallback)
 
+        if interactive:
+            visibility = self.visibility
+            visibility.pop('elements')
+
+            subAxes = plt.axes([0.91, 0.65, 0.08, 0.05*len(visibility)])
```
Comment (dccote) [inline raytracing/figure.py:558]: The position is awkward.  I would put it probably higher and use larger fonts.
Also, *no box* around!

Comment (dccote) [reply raytracing/figure.py:558]: Also, in the `__main__`, the checkbox is half the size and looks squished.

Comment (JLBegin) [reply raytracing/figure.py]: Is this ok ? 
image

image

**Comment by dccote** [comment]
I changed the labels.  I just need the figure to expand on the right and have the checkbox as a single line.

**Pull request merged**